### PR TITLE
update joblib CI scripts

### PR DESCRIPTION
**Pull request by pierreglaser**
Fix miscellaneous CI failures.

### CODE REVIEW

**Comment by pierreglaser** [comment]
The conda solver fails in an undeterministic fashion for the joblib + scikit-learn entry.

**Comment by pierreglaser** [comment]
I proposed an upstream fix to the `distributed` bit of this PR: https://github.com/dask/distributed/pull/2841

File: continuous_integration/travis/test_script.sh [hunk 1]
```diff
@@ -22,11 +22,11 @@ fi
 if [[ "$SKLEARN_TESTS" == "true" ]]; then
     # Install scikit-learn from conda and test against the installed
     # development version of joblib.
-    conda install --yes cython pillow scikit-learn scipy==1.2.1
+    python -mpip install cython==0.29 pillow==6.0 scipy==1.2.1 scikit-learn==0.20.3
```
Comment (ogrisel) [inline continuous_integration/travis/test_script.sh]: Let's test for the latest stable version of everything:

```suggestion
    python -m pip install cython pillow scipy scikit-learn
```

File: continuous_integration/travis/test_script.sh [hunk 2]
```diff
@@ -22,11 +22,11 @@ fi
...
     python -c "import sklearn; print('Testing scikit-learn', sklearn.__version__)"
     # Skip test_lars_cv_max_iter because of a warning that is (probably)
     # not related to joblib. To be confirmed once the following PR is
     # merged:
     # https://github.com/scikit-learn/scikit-learn/pull/12597
-    pytest -vl -k "not test_lars_cv_max_iter" --pyargs sklearn
+    pytest --cache-clear -vl -k "not test_lars_cv_max_iter" --pyargs sklearn
```
Comment (ogrisel) [inline continuous_integration/travis/test_script.sh]:
```suggestion
    pytest -vl -k "not test_lars_cv_max_iter" --pyargs sklearn
```

File: joblib/test/test_dask.py [hunk 3]
```diff
@@ -302,6 +303,9 @@ def test_wait_for_workers(cluster_strategy):
     finally:
         client.close()
         cluster.close()
+        del client
+        del cluster
+        gc.collect()
```
Comment (ogrisel) [inline joblib/test/test_dask.py]: Please add a comment to state for which version of distributed this is needed.

Comment (ogrisel) [reply joblib/test/test_dask.py]: And a link to the distributed PR / issue.

**Comment by ogrisel** [comment]
Thanks @pierreglaser!

**Pull request merged**